ui/hooks/src/utils/entry-utils.ts

```diff
@@ -217,11 +169,33 @@ export const createPendingEntry = (
   };
 };
 
-export const buildPublishObject = (data: any, parentEntryId?: string) => {
-  // save only the ipfs CID prepended with CID: for the slate content image urls
-  const cleanedContent = data.content.map((node: any) => {
+export interface EntryPublishObject {
+  data: PostResponse['content'];
+  post: {
+    tags: IPublishData['metadata']['tags'];
+    mentions: IPublishData['metadata']['mentions'];
+  };
+
+  quotes: IPublishData['metadata']['quote']['entryId'][];
+}
+export interface CommentPublishObject {
+  data: CommentResponse['content'];
+  comment: {
+    tags: IPublishData['metadata']['tags'];
+    mentions: IPublishData['metadata']['mentions'];
+    postID: string;
+  };
+}
+
+export function buildPublishObject(data: IPublishData): EntryPublishObject;
+export function buildPublishObject(data: IPublishData, parentEntryId: string): CommentPublishObject;
+export function buildPublishObject(data: IPublishData, parentEntryId?: string) {
```

Comment:
Why is this defined multiple times?

Reply:
it's function overloading, i am returning a different type based on that `parentEntryId` parameter.
I can remove the typings in from the actual function declaration. i've left them there because it's easier to see them in the editor's autocomplete popovers

Reply:
ok, good to know:)